packages/apps/ds4ch/pages/[slug].vue

```diff
@@ -8,39 +8,52 @@ const route = useRoute();
 const contentful = inject("$contentful");
 const { localeProperties } = useI18n();
 
-const { data: page } = await useAsyncData(
-  `landingPage:${route.params.slug}`,
-  async () => {
+const { data: contentfulResponse, error: contentfulErrors } =
+  await useAsyncData(`landingPage:${route.params.slug}`, () => {
     const variables = {
       identifier: route.params.slug,
       locale: localeProperties.value.language,
     };
 
-    const response = await contentful.query(landingPageQuery, variables);
-    return response.data?.landingPageCollection?.items?.[0];
-  },
-);
+    return contentful.query(landingPageQuery, variables);
+  });
 
-if (!page.value) {
-  showError({ statusCode: 404, statusMessage: "Not Found" });
-}
+const page = computed(() => {
+  return contentfulResponse.value?.data?.landingPageCollection?.items?.[0];
+});
 
-const sections = page.value.hasPartCollection?.items.filter((item) => !!item);
+const error = computed(() => {
+  if (
+    !contentfulErrors.value &&
+    !contentfulResponse.value?.data?.landingPageCollection?.items?.[0]
+  ) {
+    showError({ statusCode: 404, statusMessage: "Not Found" });
```

Comment:
Why sometimes show the Nuxt error page and other times show the custom alert message? Don't we always want the latter?